ext/debase_internals.c

```diff
@@ -371,7 +373,7 @@ process_line_event(VALUE trace_point, void *data)
       context->stop_frame = -1;
     }
 
-    breakpoint = breakpoint_find(breakpoints, path, lineno);
+    breakpoint = breakpoint_find(breakpoints, path, lineno, binding);
```

Comment:
Makes sense to defer, if there is a measurable performance hit.

But I do not see how the test functions will be able to pass in a `TracePoint` object.  Of course, we could cheat and have the tests pass in the `Binding` as they are doing, and have `process_line_event` pass in the `TracePoint` object.  (We're already abstracted behind the `VALUE` type, so this doesn't break typing.)  Then `check_breakpoint_expr` would verify the type of the `binding` parameter and, as needed, perform the actual `rb_tracearg_binding` call there.  But, that would be ugly mixing test-centric code with the main code.

An alternative is to use something like a promise.  (I presume Ruby has them; they seem to exist in some form or another in almost all languages.)  This results in nearly the same behavior as above, but it feels slightly less dirty.  Thoughts?